data/postsRouter.js

```diff
@@ -0,0 +1,82 @@
+const router = require("express").Router();
+
+const db = require("./db.js");
+
+//-------- New Post  --------//
+router.post("/", (req, res) => {
+  const newPost = res.body;
+  if (newPost.title && newPost.contents) {
+    db.insert(newPost)
+      .then(post => {
+        res.status(201).json(newPost);
+      })
+      .catch(err => {
+        res.status(501).json({ message: "Failed to add Post" });
+      });
+  } else {
+    res.status(400).json({
+      Message: "Please provide title and contents for the post."
+    });
+  }
+});
+
+//------Get all Posts-------//
+router.get("/", (req, res) => {
+  db.find()
+    .then(posts => {
+      res.status(202).json(posts);
+    })
+    .catch(err => {
+      res.status(501).json({ message: "Failed To Get Posts" });
+    });
+});
+
+//-------- Get Posts By Id--------//
+
+router.get("/:id", (req, res) => {
+  db.findById(req.params.id)
+    .then(post => {
```

Comment:
You want to add validation in your .then 
if (!post) {
    return res.status(404).json({err: "Error getting user"})
}